src/main/java/frc/robot/subsystems/BallTransport.java

```diff
@@ -0,0 +1,195 @@
+            topMotor.set(TRANSPORT_SPEED);
+        }).until(() -> {
+            return laserSwitch.getAsBoolean();
+        });
+
+    }
+
+    private CommandBase ballAtLaser() {
+        return cmd("Keep ball on laser switch").onExecute(() -> {
+            bottomMotor.set(TRANSPORT_SPEED);
+            topMotor.stopMotor();
+        }).until(() -> {
+            return colorSensorBallLimit();
+        });
+    }
+
+    public CommandBase stopTransport() {
+        return instant("Stop Ball Transport", () -> {
+            bottomMotor.stopMotor();
+            topMotor.stopMotor();
+        });
+    }
+
+    public CommandBase loadShooter() {
+        return cmd("Load Shooter").onExecute(() -> {
+            topMotor.set(TRANSPORT_SPEED);
+            bottomMotor.set(TRANSPORT_SPEED);
+        }).onEnd(() -> {
+            // this assumes both balls have been shot
+            colorBuffer.clear();
+            topMotor.stopMotor();
+            bottomMotor.stopMotor();
+        }).withTimeout(4);
+    }
+
+    private Command noBall = noBall();
+    private Command ballAtColor = ballAtColor();
+    private Command ballAtLaser = ballAtLaser();
+    private Command ballAtLaserAndColor = stopTransport();
+
+    // command selector enum used for command selector
+    private enum CommandSelector {
+        WAITFORBALL,
+        MOVEBALLTOLASER,
+        WAITFORBALLNOLASER,
+        INTAKEFILLED
+    }
+
+    // returns a value of command selector enum based on sensor input.
+
+    private CommandSelector commandSelector() {
+        if (colorSensorBallLimit() && laserSwitch.getAsBoolean()) {
+            return CommandSelector.WAITFORBALL;
+        } else if (colorSensorBallLimit()) {
+            return CommandSelector.MOVEBALLTOLASER;
+        } else if (laserSwitch.getAsBoolean()) {
+            return CommandSelector.WAITFORBALLNOLASER;
+        } else {
+            return CommandSelector.INTAKEFILLED;
+        }
+    }
+
+    // Creates a map of entries for the command selector to use.
+    private final Map<Object, Command> selectCommandMap = Map.ofEntries(
+            Map.entry(CommandSelector.WAITFORBALL, noBall),
+            Map.entry(CommandSelector.MOVEBALLTOLASER, ballAtLaserAndColor),
+            Map.entry(CommandSelector.WAITFORBALLNOLASER, ballAtColor),
+            Map.entry(CommandSelector.INTAKEFILLED, ballAtLaser));
+
+    // select command that determines what command needs to be run based on the
+    // command selector.
+    // this needs to be run with intake
+    public SelectCommand loadCargoWithIntake() {
+        return new SelectCommand(selectCommandMap, this::commandSelector);
+    }
+
+    // Unloads cargo that is opposite of the alliance color
+    // Intake must be deployed backwards for this
+    public CommandBase removeCargo() {
+        return cmd("Remove \"Wrong Colored\" Cargo").onExecute(() -> {
+            switch (allianceColor) {
+                default:
```

Comment:
`default` case should always come after the other cases - the `switch` goes into the first branch that it matches. When that is `default`, then everything will trigger that case.